src/NUnitConsole/nunit3-console/TestEventHandler.cs

```diff
@@ -98,6 +103,8 @@ private void TestFinished(XmlNode testResult)
             var status = testResult.GetAttribute("label") ?? testResult.GetAttribute("result");
             var outputNode = testResult.SelectSingleNode("output");
 
+            FlushNewLineIfNecessary(testName);
+
```

Comment:
Same question as above: in this case a test has ended. Depending on the settings, it may or may not be about to write anything. If it isn't going to write anything, then it's still possible that the test with a pending newline may finish the line before any other test produces output.